### PR TITLE
Navigation - Simplify form code, separate paths

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
A little cleanup to pave the way for #31520

Technical Details
--------

This follows the pattern we've been doing while converting pages to AdminUI: detangle the 'page' from the 'form' so they don't share the same path.

### CODE REVIEW

**Comment by ufundo** [comment]
Looks great, thanks @colemanw

**Comment by ufundo** [comment]
R-run on test site - I can still:
- open the menu editor
- add new items using the top button
- add new sub items using
- edit existing items

I'm conscious people could have linked the create/edit forms externally ( like `/civicrm/admin/menu?reset=1&action=add` ) , and they will now just receive the top menu rather than the form? But it should be fairly clear what's happened, so hopefully not too painful.

File: schema/Core/Navigation.entityType.php [hunk 1]
```diff
@@ -11,6 +11,12 @@
     'add' => '3.0',
     'label_field' => 'label',
   ],
+  'getPaths' => fn() => [
+    'browse' => 'civicrm/admin/menu',
+    'add' => 'civicrm/admin/menu/item?reset=1&action=add',
+    'update' => 'civicrm/admin/menu/item?reset=1&action=update&id=[id]',
+    'delete' => 'civicrm/admin/menu/item?reset=1&action=delete&id=[id]',
```
Comment (ufundo) [inline schema/Core/Navigation.entityType.php:18]: @colemanw just noticed the delete dialog isn't actually using this form and this link doesn't seem to work for me

Comment (colemanw) [reply schema/Core/Navigation.entityType.php:18]: @ufundo yea the jstree form wasn't using it either, so maybe it never actually worked even though the code stub is there in the form.

**Pull request merged**